### ISSUE TITLE
uglification: Unknown provider: $futureStateProviderProvider

### ISSUE THREAD

**Issue by stephanMettler** [christopherthielen/ui-router-extras#266]
Unknown provider: $futureStateProviderProvider <- $futureStateProvider <- $futureState

Please help!
this code works fine, but fails with the above error after minification:

I tried this with a fresh yeoman build of angular-fullstack and added the following code:
```
angular.module('futureApp', [
	'ngResource',
	'ui.router',
	'ct.ui.router.extras',
	'ct.ui.router.extras.future'
	])
.config(['$futureStateProvider', function($futureStateProvider) {
	var loadAndRegisterFutureStates = function ($http) {
		return $http.get("future.js").then(function (resp) {
			$futureStateProvider.futureState( {
				"stateName": "iframe2",
				"urlPrefix": "/iframe2",
				"type": "iframe",
				"src": "/someHtml.html"
			});
		});
	};
    $futureStateProvider.addResolve(loadAndRegisterFutureStates);
}])

```
Updating to grunt-ng-annotate to 1.0.2 did not help me!
updated all my packages in package.json to latest version

Even tried injection like this:
```
loadAndRegisterFutureStates.$inject = ['$futureStateProvider'];
```

I have to disable mangling to make it work ;(
```
 uglify: {
    	options: {
    		report: 'min',
    		mangle: false
    	}
    },
```

**Comment by gibbitz** [comment]
I thought I was the only one. Having the exact same issue. It looks like the service initialization may be the issue.
`
 // inject $futureState so the service gets initialized via $get();
  app.run(['$futureState', function ($futureState, $rootScope) {
    statesAddedQueue.itsNowRuntimeOhWhatAHappyDay($rootScope);
  } ]);
`
should be:
`
 // inject $futureState so the service gets initialized via $get();
  app.run(['$futureState', ['$futureState', '$rootScope', function ($futureState, $rootScope) {
    statesAddedQueue.itsNowRuntimeOhWhatAHappyDay($rootScope);
  }] ]);
`
Though I would think using the minified version would blow up regardless of whether my app.js is obfuscated or not. I'm not finding that to be the case.

**Comment by nkanand4** [comment]
@stephanMettler
I was running into the same issue and found out that if I am using $futureStateProvider.addResolve, the function passed to it should be annotated. i.e. $futureStateProvider.addResolve(['$http', '$log', function ($http, $log) { }]) and then the minification worked just fine.

**Comment by critical** [comment]
Hi @nkanand4,
I have the same issue during uglify when I try to add a state factory with the following code
```
$futureStateProvider.stateFactory('ocLazyLoad', ['$q', '$ocLazyLoad', function($q, $ocLazyLoad, futureState) {  
  var deferred = $q.defer();  
  $ocLazyLoad.load(futureState.module).then(function(name) {
      deferred.resolve();
     }, function() {
      deferred.reject();
    }
  );
  return deferred.promise;
}]);
``` 
The error is `Cannot read property 'module' of undefined` because the parameter _futureState_ is undefined. What am I doing wrong?

**Comment by nkanand4** [comment]
Look at the first line, it seems you are missing to annonate futureState inline.

**Comment by critical** [comment]
Yes because _futureState_ is the parameter passed  by ui-router-extras (line 1435). I can't inject this value.
I'm using angular v.: 1.5.5, ui-router v.: 0.2.18 and ui-router-extras v.: 0.1.2.
Thank you

**Comment by critical** [comment]
I resolved the problem. It was a my fault.
Many thanks.
